creator/data_templates/field_definitions_schema.py

```diff
@@ -0,0 +1,243 @@
+    """
+    if input_value is None:
+        return input_value
+
+    i_value = None
+    f_value = None
+    try:
+        f_value = float(input_value)
+        i_value = int(input_value)
+    except ValueError:
+        pass
+
+    # Input was a non-convertible str
+    if (i_value is None) and (f_value is None):
+        new_value = input_value
+    # Input was an int
+    elif i_value == f_value:
+        new_value = i_value
+    # Input was a float
+    else:
+        new_value = f_value
+
+    return new_value
+
+
+def coerce_bool(input_value: str = None) -> Union[bool, str]:
+    """
+    Try to coerce a str literal to a boolean
+    Coercion will include attempting to translate str synonyms to booleans
+
+    If coercion is not possible return the original input value
+    """
+    input_value = coerce_number(input_value)
+
+    if isinstance(input_value, str):
+        alternates = {
+            "true": True,
+            "false": False,
+            "yes": True,
+            "no": False,
+            "required": True,
+            "not required": False,
+        }
+        new_value = alternates.get(input_value.lower(), input_value)
+    else:
+        try:
+            new_value = bool(input_value)
+        except ValueError as e:
+            new_value = input_value
+
+    return new_value
+
+
+class FieldDefinitionSchema(Schema):
+    """
+    Marshmallow schema for a cleaning and validating a JSON object
+    containing a field definition instance in TemplateVersion.field_definitions
+
+    Notes:
+        The missing kwarg is used when the marshmallow field key is not present
+        in the input data and marshmallow needs to know if it should insert
+        the key with some kind of value
+
+        The allow_none kwarg is used when you want to allow the field to take
+        on a None value
+    """
+    key = fields.String(
+        description="Canonical name or identifier for the field",
+        missing=None,
+        allow_none=None
+    )
+    label = fields.String(
+        description="Human friendly name for the field",
+        required=True
+    )
+    data_type = fields.String(
+        description="The data type of this field's values",
+        validate=validate.OneOf(
+            ['enum', 'string', 'number', 'boolean', 'date']
+        ),
+        missing="string"
+    )
+    required = fields.Boolean(
+        description="Whether the field is required to be populated",
+        missing=False
+    )
+    accepted_values = fields.List(
+        fields.Raw(),
+        description="The list of acceptable values this field may be"
+                    "populated with",
+        # This says, if accepted_values is not None, then make sure it has
+        # at least one member in the list
+        validate=validate.Length(min=1),
+        missing=None,
+        allow_none=True
+    )
+    missing_values = fields.List(
+        fields.String(allow_none=True),
```

Comment:
why `fields.String` and not `fields.Raw`?

Reply:
I think we always want missing values to be represented by strings (e.g. Not Available, Unknown). I can't think of why we would want to accept any other data type. For example, if i had an integer field like Age at Diagnosis and I allowed integers for missing values then how do I distinguish between real integer values vs the missing value which is also some sort of integer. For this reason, I think its best to accept only strings or null which is represented by None

Reply:
thanks for explaining!